lib/please_cli/please_cli/config.py

```diff
@@ -59,9 +58,9 @@
 }
 
 DEV_PROJECTS = ['postgresql']
-PROJECTS = list(map(lambda x: x.replace('_', '-'),
+PROJECTS = list(map(lambda x: x.replace('_', '-')[len(SRC_DIR) + 1:],
                     filter(lambda x: os.path.exists(os.path.join(SRC_DIR, x, 'default.nix')),
-                           os.listdir(SRC_DIR))))
+                           glob.glob(SRC_DIR + '/*') + glob.glob(SRC_DIR + '/*/*'))))
```

Comment:
initially i also want this for any level, but only did it for max depth 2 for few reasons:
1. performance, accidentally i had node_modules in one of my projects and it every ./please command was taking quite some time. we would either have to do it lazily with some caching.
2. i would have to implement this also in nix, which i didn't want to do since we will remove this code once we start working on #1220 
3. I think having this done for 2 levels unblocks #1294 while not being too intrusive

In gist, once we start working on #1220 project resolution will be moved from current implementation to only being done in please_cli (eg. one source of truth) and not in 2 places like now. also if we in future want to support "remote" projects (eg. projects living outside `release-services`) then we would have to implement this in a performant way.